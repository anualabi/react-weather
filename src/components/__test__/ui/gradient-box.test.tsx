import { render, screen } from "@testing-library/react";

import GradientBox, { GradientBoxProps } from "@/components/ui/gradient-box";

const mockProps = {
  children: <div>children</div>,
  className: "",
};

const customRender = (props: GradientBoxProps) => {
  return render(<GradientBox {...props} />);
};

describe("GradientBox", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = customRender({ ...mockProps });
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render children", () => {
    customRender({ ...mockProps });
    expect(screen.getByText("children")).toBeInTheDocument();
  });
});
