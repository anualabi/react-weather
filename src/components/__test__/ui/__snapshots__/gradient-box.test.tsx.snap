// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientBox should render (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="bg-gradient-to-b from-transparent from-0% via-[#C4C4C405] via-43.85% to-[#C4C4C420] to-100% rounded-md"
  >
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;
